### DISCUSSION TITLE
Grid layout, full height content

### DISCUSSION THREAD

**Discussion by 2acode** [tailwindlabs/tailwindcss#11796] [Help] [answered]
Hi,
i'm trying to create a simple layout with "header" and "main" using "grid" instead of "flex".
With this code I expected the "main" to expand to the bottom of the page but it doesn't.

```
<div class="border-4 border-lime-500 h-screen">
    <div class="grid grid-cols-1 grid-rows-[auto_1fr]">
        <header class="border border-blue-500">
            header
        </header>

        <main class="border border-red-500 overflow-auto">
            {{ $slot }}
        </main>
    </div>
</div>
```

![grid-tw](https://github.com/tailwindlabs/tailwindcss/assets/13646003/2534086c-166a-4951-9a1e-13e18a206e4b)

Using CSS (no Tailwind) with grid-template-areas and grid-template-rows the result is correct.

`grid-template-rows: auto 1fr;`

What am I doing wrong?
Thank you

**Comment by mybearworld** [comment]
Flex would probably work better here: https://play.tailwindcss.com/XIFvGG2Jox

**Reply by 2acode** [reply]
Thank you,
however, I already have all the layouts with "flex" in TW so now I'm experimenting with "grid" to test new solutions.

**Comment by nikolailehbrink** [comment, marked as answer]
Hey there @2acode, I just tested it without the tailwind class and it didn't work for me with `grid-template-rows: auto 1fr`. That said you have to set the height on your grid aswell, that makes it work:

```html
<div class="h-screen border-4 border-lime-500">
  <div class="grid h-full grid-rows-[auto_1fr]">
    <header class="border border-blue-500">header</header>
    <main class="overflow-auto border border-red-500">{{ $slot }}</main>
  </div>
</div>
```

Note: You don't have to set `grid-cols-1`

**Reply by 2acode** [reply]
Thank you!
I thought it wasn't necessary with grid, my mistake.

Therefore this solution is fine too, without using the inner div:

```
<div class="h-screen border-4 border-lime-500 grid grid-rows-[auto_1fr]">
    <header class="border border-blue-500">
        header
    </header>
    
    <main class="overflow-auto bg-red-500">
        content
    </main>
</div>
```

Regarding the solution with CSS only here is the working code:

```
<!doctype html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Grid</title>
    <meta name="description" content="Grid">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style>
        html, body {
            padding: 0;
            margin: 0;
        }
        #layout {
            display: grid;
            grid-template-areas:
                'header'
                'main';
            grid-template-rows: auto 1fr;
            height: 100vh;
        }
        #header {
            grid-area: header;
        }
        #main {
            grid-area: main;
            background-color: gray;
        }
    </style>
</head>

<body>
    <div id="layout">
        <header id="header">
            header
        </header>

        <main id="main">
            main
        </main>
    </div>
</body>
</html>
```
